compiler/fir/analysis-tests/testData/extendedCheckers/VariableAssignmentChecker.kt

```diff
@@ -0,0 +1,135 @@
+// WITH_RUNTIME
+
+import kotlin.properties.Delegates
+
+fun withAnnotation(p: List<Any>) {
+    @Suppress("UNCHECKED_CAST")
+    <!CAN_BE_VAL!>var<!> v = p as <!REDUNDANT_EXPLICIT_TYPE!>List<String><!>
+    print(v)
+}
+
+fun withReadonlyDeligate() {
+    val s: String by lazy { "Hello!" }
+    s.hashCode()
+}
+
+fun destructuringDeclaration() {
+    var (<!CAN_BE_VAL!>v1<!>, <!CAN_BE_VAL!>v2<!>) = getPair()
+    print(v1)
+
+    var (<!CAN_BE_VAL!>v3<!>, v4) = getPair()
```

Comment:
We should report `CAN_BE_VAL` on destructuring declaration iff *all* its components can be val. Please think how to achieve this.

Reply:
Ideally would be to report `CAN_BE_VAL` directly on the `var` keyword, not on components